web/src/click_handlers.js

```diff
@@ -774,13 +774,15 @@ export function initialize() {
         e.preventDefault();
         e.stopPropagation();
 
-        if (!$(e.target).hasClass("direct-messages-list-filter")) {
-            // Avoiding having clicks on the filter input.
-            //
-            // TODO: Refactor to use more precise selectors for this
-            // click handler in general; this is a fragile pattern.
-            window.location.hash = "narrow/is/dm";
-        }
+        window.location.hash = "narrow/is/dm";
+    });
+
+    $("body").on("click", ".direct-messages-list-filter", (e) => {
+        // We don't want clicking on the filter to trigger the DM
+        // narrow defined on click for
+        // `#direct-messages-section-header.zoom-in`.
+        e.preventDefault();
+        e.stopPropagation();
```

Comment:
Does this prevent moving your cursor when clicking in the filter input? We might want just `stopPropagation` to avoid that.

Reply:
It did not, but I've removed `e.preventDefault` since it wasn't affecting any behaviour here.

Reply:
Cool, yeah it's good to be thoughtful about which of those one needs and why; we've had a lot of bugs in the past from people copy-pasting both in when they actually mean just one or the other.